src/sync/sync_manager.cpp

```diff
@@ -411,6 +411,32 @@ void SyncManager::set_current_user(const std::string& user_id)
     m_metadata_manager->set_current_user_identity(user_id);
 }
 
+void SyncManager::remove_user(const std::string& user_id)
+{
+    std::lock_guard<std::mutex> lock(m_user_mutex);
+
+    auto it = std::find_if(m_users.begin(),
+                           m_users.end(),
+                           [user_id](const auto& user) {
+        return user->identity() == user_id;
+    });
+    
+    if (it == m_users.end())
+        return;
+    
+    auto user = *it;
+    
+    if (!m_metadata_manager) {
+        return;
+    }
+    
+    if (user->identity() == m_metadata_manager->get_current_user_identity()) {
+        m_metadata_manager->set_current_user_identity("");
+    }
+    
+    m_users.erase(it);
```

Comment:
This looks dangerous. SDK's might still have a pointer to the User after it has been removed. I suspect doing this will cause all operations to fail after this. See https://mongodb.slack.com/archives/CKW2HG3M0/p1584430859364900 for more context.

IMO it would make sense to rework the current User states, so instead of the current ones we have `LOGGED_IN, LOGGED_OUT and REMOVED. It seems to fit the semantics we offer in a better way.

Reply:
Changed, also I have updated the `SyncUser::State` enum to reflect your suggestion